src/views/signup.test.js

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import axios from 'axios';
+import { fireEvent, render, screen, wait } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import SignupForm from "./signup";
+import UserContext from "../UserContext";
+
+const mockHistory = { push: jest.fn() };
+
+const mockNotAuthenticatedUser = {
+  isAuthenticated: false,
+};
+
+const mockInputEvent = { target: {value: "mock value"} };
+const mockInputEmailEvent = { target: {value: "mock@mockdomain.com"} };
+
+describe("signup component", () => {
+  let container = null;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.spyOn(axios, 'post').mockResolvedValue({});
+    jest.spyOn(window, 'alert').mockImplementation(() => {});
+    container = render(
+      <UserContext.Provider value={{ user: mockNotAuthenticatedUser }}>
+        <SignupForm history={mockHistory} />
+      </UserContext.Provider>
+    );
+  });
+
+  it("should display an error when phone not populated", async() => {
```

Comment:
Really stellar testing coverage here, nice work! Since this specific test is essentially testing that Yup is working as expected, is this test actually needed? Open to feedback here, as I'm not a testing guru :)

Reply:
To be pure, it would be best to mock out Yup. But, it does seem valuable to test whether an Error message displays or not. Another option might be to test that the API is NOT called as a result of this button press.